crates/catalog/rest/src/catalog.rs

```diff
@@ -617,7 +617,13 @@ impl RestCatalog {
             props.extend(config);
         }
 
-        let file_io = match self.config.warehouse.as_deref().or(metadata_location) {
+        let warehouse_path = match self.config.warehouse.as_deref() {
+            Some(url) if url.contains("://") => Some(url),
```

Comment:
How about replacing it with [Url::parse](https://docs.rs/url/latest/url/struct.Url.html#method.parse)? It seems more robust to me.

Reply:
This is also happening [below in the `FileIO::from_path`](https://github.com/apache/iceberg-rust/blob/d57d91b9a72c516c6665d5faef349f52ebe59585/crates/iceberg/src/io.rs#L155), but it raises an exception we want to avoid.

Reply:
Yes, we don't need to raise an exception here, just match `if Url::parse().is_ok()` would be enough.

Reply:
Ah, I see. Love it 👍